File: 0069_Sqrt_x/Solution.java
public class Solution {
	public int mySqrt(int x) {
		if (x <= 1) {
			return x;
		}

		int start = 0;
		int end = x;
		int mid;
		while (start != end - 1) {
			mid = (start + end) / 2;
			if (mid == x / mid) {
				return mid;
			}
			// if mid is too big
			else if (x / mid < mid) {
				end = mid;
			} else {
				start = mid;
			}
		}
		// sqrt(2) 1.414 hence 1
		return start;
	}
}

File: 0069_Sqrt_x/Solution.py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x < 2:
        	return x

        start = 0;
        end = x;
        while start != end-1:
        	mid = (start + end) / 2
        	if x/mid == mid:
        		return mid
        	elif x/mid < mid:
        		end = mid
        	else:
        		start = mid

        # sqrt(2) 1.414 hence 1
        return start
